kaiax/staking/staking_info.go

```diff
@@ -125,13 +146,45 @@ func (si *StakingInfo) consolidateNodes() *[]consolidatedNode {
 		}
 	}
 
+	// CLStakingInfo can only exist after Prague HF.
+	if len(si.CLStakingInfos) > 0 {
+		for _, clsi := range si.CLStakingInfos {
+			// If the nodeId of CLStakingInfo is not found in nToR, it means the validator is not in the AddressBook.
+			// So we skip it.
+			if r, ok := nToR[clsi.CLNodeId]; ok {
+				// One CLStakingInfo per validator is guaranteed by CLRegistry.
+				cmap[r].CLStakingInfo = clsi
+			}
+		}
+	}
+
 	carr := make([]consolidatedNode, 0, len(cmap))
 	for _, r := range rList {
 		carr = append(carr, *cmap[r])
 	}
 	return &carr
 }
 
+func (c consolidatedNode) Split(amount *big.Int) (*big.Int, *big.Int) {
```

Comment:
Move it to reward module please.

Reply:
Then I think moving `consolidatedNode` to reward module and making it public would be better. Do you prefer to leave it in a staking module?

Reply:
I prefer having consolidatedNode in staking module because it's used by both reward and valset modules.
Looking at the code, it seems difficult to separate this Split function from staking model. Let's just keep it here.